nutpie/compile_stan.py

```diff
@@ -41,67 +110,33 @@ def compile_stan_model(
         with open(filename, "r") as file:
             code = file.read()
 
-    model_id = httpstan.models.calculate_model_name(code)
-    stan_lib = None
-    if cache and not extra_compile_args:
-        try:
-            stan_lib = httpstan.models.import_services_extension_module(model_id)
-        except (FileNotFoundError, KeyError):
-            pass
-    if stan_lib is None:
-        output = httpstan.models.build_services_extension_module_sync(code, extra_compile_args)
-        if show_compiler_output:
-            print(output)
-        stan_lib = httpstan.models.import_services_extension_module(model_id)
-
-    ctx = stan_lib.new_logp_ctx(data)
-    n_dim = stan_lib.num_unconstrained_parameters(ctx)
-
-    shape_info = _make_shape_info(stan_lib, data)
-
-    logp_maker = _make_logp_maker(stan_lib, data)
-
-    def expanding_function(x, seed, chain, draw):
-        return stan_lib.write_array_ctx(ctx, x, True, True, seed + 10000 * chain + draw)
-
+    if model_name is None:
+        model_name = "model"
+
+    with tempfile.TemporaryDirectory() as basedir:
+        model_path = (
+            pathlib.Path(basedir)
+            .joinpath("name")
+            .with_name(model_name)  # This verifies that it is a valid filename
+            .with_suffix(".stan")
+        )
+        model_path.write_text(code)
+        make_args = ["STAN_THREADS=true"]
+        if extra_compile_args:
+            make_args.extend(extra_compile_args)
+        so_path = bridgestan.compile_model(model_path, make_args=make_args)
+        # Set necessary library loading paths
+        bridgestan.compile.windows_dll_path_setup()
+        library = lib.StanLibrary(so_path)
+
+    # One the library is loaded we can delete the temporary dir
+    # TODO: Is this also true on Windows?
```

Comment:
In my experience no, you're not able to delete loaded DLLs

Reply:
Yes, seems windows locks files that are opened...
On unix the file just decreases it refcount and only really disappears once all programs close it.
I guess I'll have to put that in a destructor somewhere...